lib/Sema/CSGen.cpp

```diff
@@ -355,7 +365,233 @@ namespace {
     /// Ignore types.
     bool walkToTypeLocPre(TypeLoc &TL) override { return false; }
   };
-  
+
+  bool expandCollectedTypesFromOperatorOverloads(ConstraintSystem &CS,
+                                                 LinkedTypeInfo &lti) {
```

Comment:
I'm actually hoping that we could remove all of this code from CSGen instead of adding more :)